lib/wrap-request.js

```diff
@@ -0,0 +1,38 @@
+module.exports = wrapRequest
+
+const noop = () => Promise.resolve()
+
+function wrapRequest (octokit, emitter, globalLimiter, writeLimiter, request, options) {
+  return globalLimiter.schedule(async function () {
+    if (options.method !== 'GET') {
+      // Assuming all non-GET calls are WRITEs
+      await writeLimiter.schedule(noop)
+    }
+
+    try {
+      const response = await request(options)
+      return response
+    } catch (error) {
+      if (error.status === 403 && error.message.toLowerCase().includes('abuse')) {
+        // The user has hit the abuse rate limit.
+        // https://developer.github.com/v3/#abuse-rate-limits
+
+        // Due to known issues with replication lag, the Retry-After header can sometimes be blank
+        // for a few seconds after entering 'abuse limited mode'.
+        // We ensure the Retry-After is always at least 5 seconds.
+        const retryAfter = Math.max(~~error.headers['retry-after'], 5)
```

Comment:
Do you use `~~` as a shortcut for `parseInt(..., 10)`? I wonder if the latter would be more readable? In that case I think we would do 

```suggestion
        const retryAfter = Math.max(parseInt(error.headers['retry-after'] || 0, 10), 5)
```

I’m good either way, the comment above makes it clear anyway, just curious

Reply:
Yes `~~something` does the same as `parseInt(something, 10)`, except it also safely returns `0` for `null`, `undefined`, `''`, etc. `parseInt()` blows up with an exception, so `~~` is perfect in this case since I wrap it in `Math.max()`.